lib/builtins/deploy-delegates/lambda-deployer/helper.js

```diff
@@ -232,13 +248,48 @@ function _addEventPermissions(lambdaClient, skillId, functionArn, profile, callb
     });
 }
 
+function _getLambdaCustomValue(alexaRegion, userConfig, parameter) {
+    return R.path(['regionalOverrides', alexaRegion, 'lambda', parameter], userConfig)
+    || R.path(['lambda', parameter], userConfig);
+}
+
+function _getLambdaFunctionName(alexaRegion, userConfig, skillName, profile) {
+    // Generates a valid Lambda function name.
+    //  a lambda function name should follow the pattern: ask-skillName-profileName-alexaRegion-timeStamp
+    //  a valid function name cannot longer than 64 characters, so cli fixes the project name no longer than 22 characters
+    const generateFunctionName = () => {
+        const validSkillName = stringUtils.filterNonAlphanumeric(skillName.toLowerCase()).substring(0, 22);
+        const validProfile = stringUtils.filterNonAlphanumeric(profile.toLowerCase()).substring(0, 15);
+        return `ask-${validSkillName}-${validProfile}-${alexaRegion}-${Date.now()}`;
+    };
+
+    return _getLambdaCustomValue(alexaRegion, userConfig, 'functionName') || generateFunctionName();
+}
+
+function _getLambdaFunctionConfig(alexaRegion, userConfig) {
+    const { runtime, handler } = userConfig;
+    const description = _getLambdaCustomValue(alexaRegion, userConfig, 'description');
+    const memorySize = _getLambdaCustomValue(alexaRegion, userConfig, 'memorySize');
+    const timeout = _getLambdaCustomValue(alexaRegion, userConfig, 'timeout');
+    const environmentVariables = _getLambdaCustomValue(alexaRegion, userConfig, 'environmentVariables');
+    return {
+        runtime,
+        handler,
+        description,
+        // default values to account for Java runtime cold start
+        memorySize: Number.parseInt(memorySize, 10) || 512,
+        timeout: Number.parseInt(timeout, 10) || 15,
```

Comment:
nit: Should we put the memorySize and timeout in the constants file? (I know this is just moved from pre-existing code)